freva-rest/src/freva_rest/config.py

```diff
@@ -440,6 +397,33 @@ def mongo_collection_userdata(self) -> AsyncIOMotorCollection:
             self.mongo_client[self.mongo_db]["user_data"],
         )
 
+    @property
+    def mongo_collection_flavours(self) -> AsyncIOMotorCollection:
+        """Define the mongoDB collection for custom flavours."""
+        return cast(
+            AsyncIOMotorCollection,
+            self.mongo_client[self.mongo_db]["custom_flavours"],
+        )
+
+    # TODO: the Logik of this function needs to be brainstormed.
+    # First and naive solution
+    def is_admin_user(self, current_user: BaseModel) -> bool:
+        """Return True if the user holds the 'realm-admin' (or 'admin') role."""
+        if (
+            hasattr(current_user, 'preferred_username')
+            and current_user.preferred_username in self.admin_list
+        ):
+            return True
+
+        claims = current_user.dict()
+        roles = (
+            claims
+            .get("resource_access", {})
+            .get("realm-management", {})
+            .get("roles", [])
+        )
+        return "realm-admin" in roles or "admin" in roles
```

Comment:
Either you make this around a predefined list of admins users and check if the current token user name in the admin list *or* you make this based on token claims. Then instead to setting admin user names you set a regex for the token claims and match it here. The advantage is, that once set you can go to keycloak and add users there. No restart, no redeployment necessary. Hence I'd prefer that method.